rclrs/src/parameter/value.rs

```diff
@@ -319,6 +324,107 @@ impl ParameterVariant for ParameterValue {
     }
 }
 
+impl From<ParameterValue> for RmwParameterValue {
+    fn from(value: ParameterValue) -> Self {
+        match value {
+            ParameterValue::Bool(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_BOOL,
+                bool_value: v,
+                ..Default::default()
+            },
+            ParameterValue::Integer(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_INTEGER,
+                integer_value: v,
+                ..Default::default()
+            },
+            ParameterValue::Double(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_DOUBLE,
+                double_value: v,
+                ..Default::default()
+            },
+            ParameterValue::String(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_STRING,
+                string_value: v.into(),
+                ..Default::default()
+            },
+            ParameterValue::ByteArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_BYTE_ARRAY,
+                byte_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::BoolArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_BOOL_ARRAY,
+                bool_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::IntegerArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_INTEGER_ARRAY,
+                integer_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::DoubleArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_DOUBLE_ARRAY,
+                double_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::StringArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_STRING_ARRAY,
+                string_array_value: v
+                    .iter()
+                    .map(|v| v.clone().into())
+                    .collect::<Vec<_>>()
+                    .into(),
+                ..Default::default()
+            },
+        }
+    }
+}
+
+/// An error that occured when trying to convert a parameter from an
+/// `rcl_interfaces::msg::ParameterValue`
+pub enum RmwParameterConversionError {
+    /// The parameter type was not valid.
+    InvalidParameterType,
+}
+
+impl TryFrom<RmwParameterValue> for ParameterValue {
+    type Error = RmwParameterConversionError;
+
+    fn try_from(param: RmwParameterValue) -> Result<Self, Self::Error> {
+        // TODO(luca) how to deal with PARAMETER_NOT_SET? Should we allow service calls to unset
```

Comment:
Since we support "optional" parameters, I think it makes sense to allow parameter values to be unset by a service call. But I'm not sure if there's guidance from the ROS 2 documentation on what behavior is actually expected. Also supporting that would diminish some of the elegance we get with `TryFrom` here.